liblll/CodeFragment.cpp

```diff
@@ -113,6 +113,22 @@ CodeFragment::CodeFragment(sp::utree const& _t, CompilerState& _s, bool _allowAS
 	}
 }
 
+namespace
+{
+bool validFunctionalInstruction(string us)
```

Comment:
This function assumes the `us` is a valid instruction name. This precondition should be at least put in a comment.

Alternatively, you can handle also invalid instructions and reduce the condition in `else if (c_instructions.count(us) && validFunctionalInstruction(us))`.

Reply:
Handle as in? They do not make sense to be exposed as macros (aka "functions").

Reply:
Handle as `if (it == c_instructions.end()) return false;`.

Reply:
Well, it can be added, though the callee does that already (via `count(us)`). A comment will not hurt anyway.

Reply:
I'm suggesting to actually remove `count(us)` on the callee side. Use only `validFunctionalInstruction()`.

Reply:
For readability purposes I'm not sold to remove it from the caller side.

Reply:
But did simplify it according to suggestions.